environment/k8s_environment.go

```diff
@@ -169,6 +177,69 @@ func (env *k8sEnvironment) TearDown() {
 	log.Error().Err(err)
 }
 
+// On a test failure, dump all the pod logs
+func (env *k8sEnvironment) logFailures() {
+	// Create logs directory if non-existent
+	log.Info().Msg("Creating log files on test failure")
+	logsFolder := tools.ProjectRoot + "/logs/"
```

Comment:
I'd probably also put the creation of the logs folder within TearDown in actions as this would be a common function amongst all implementations of environments.